contracts/src/v0.8/shared/test/call/CallWithExactGas.t.sol

```diff
@@ -216,6 +220,56 @@ contract CallWithExactGas__callWithExactGasSafeReturnData is CallWithExactGasSet
     assertGt(gasUsed, 500);
   }
 
+  function testFuzz_CallWithExactGasSafeReturnData_ConsumeAllGas_Success(uint8 gasLimitMultiplier) external {
+    vm.assume(gasLimitMultiplier > 0); // Assume not zero to avoid zero gas being passed to s_gasConsumer
+    uint16 maxRetBytes = 0;
+
+    vm.expectCall(address(s_gasConsumer), abi.encodeWithSelector(s_gasConsumer.consumeAllGas.selector));
+
+    // Default gas limit of 20k times the amount generated by the fuzzer
+    uint256 gasLimit = DEFAULT_GAS_LIMIT * gasLimitMultiplier;
+
+    (bool success, , uint256 gasUsed) = s_caller.callWithExactGasSafeReturnData(
+      abi.encodePacked(s_gasConsumer.consumeAllGas.selector),
+      address(s_gasConsumer),
+      gasLimit,
+      DEFAULT_GAS_FOR_CALL_EXACT_CHECK,
+      maxRetBytes
+    );
+
+    // Was determined as the least amount of gas needed for pre-call stack management.
+    // when an OOG error is thrown, used gas returned by the library is 122 <= x <= 137 based on the
+    // gas limit passed in. This was determined through trial and error and fuzzing.
+    uint256 CALL_WITH_EXACT_GAS_SAFE_RETURN_DATA_GAS_OVERHEAD = 122;
+
+    assertTrue(success, "Error: External Call Failed");
+
+    //Assert equal within a margin of error of 1/64 of the gas limit to account for excess gas used by execution library
+    assertApproxEqAbs(
+      gasUsed - CALL_WITH_EXACT_GAS_SAFE_RETURN_DATA_GAS_OVERHEAD,
+      gasLimit,
+      gasLimit / 64,
+      "Error: All gas not consumed by receiver"
+    );
+  }
+
+  function test_callWithExactGasSafeReturnData_ThrowOOGError_Revert() external {
+    uint16 maxRetBytes = 0;
+
+    vm.expectCall(address(s_gasConsumer), abi.encodeWithSelector(s_gasConsumer.throwOutOfGasError.selector));
+
+    (bool success, bytes memory retData, ) = s_caller.callWithExactGasSafeReturnData(
+      abi.encodePacked(s_gasConsumer.throwOutOfGasError.selector),
+      address(s_gasConsumer),
+      DEFAULT_GAS_LIMIT,
+      DEFAULT_GAS_FOR_CALL_EXACT_CHECK,
+      maxRetBytes
+    );
+
+    assertFalse(success, "Error: External Call Succeeded where it should not");
+    assertEq(retData.length, 0, "retData should be zero for OOG error");
```

Comment:
Since maxRetBytes is zero, this doesn't actually test anything. If anything was returned it would have been cut off

Reply:
The gas consumer contract which is invoked explicitly does not return anything when called so it doesn't make sense to include a max return bytes for this contract. 

```
        if lt(gas(), 60) {
          return(0x0, 0x0) // Return with no return data
        }
```